File: internal/domain/letter/letter.go
package letter

import "time"

type Direction string

const (
	DirectionIncoming Direction = "incoming"
	DirectionOutgoing Direction = "outgoing"
)

func (d Direction) IsValid() bool {
	return d == DirectionIncoming || d == DirectionOutgoing
}

type Letter struct {
	ID           string
	Direction    Direction
	LetterNumber int64

	DisplayLetterNumber *string

	LetterYear       *int
	LetterYearSuffix *string
	LetterSerial     *int64

	Title      string
	LetterDate time.Time

	RegistrarName string
	Sender        string
	Receiver      string

	Description *string

	CreatedBy string
	UpdatedBy *string
	DeletedBy *string

	IsDeleted bool

	CreatedAt time.Time
	UpdatedAt time.Time
	DeletedAt *time.Time
}

File: web/types/letter.ts
export type Letter = {
  id: string;
  letter_number: number;
  formatted_letter_number: string;

  title: string;
  letter_date: string;
  letter_date_jalali: string;

  registrar_name: string;
  sender: string;
  receiver: string;

  description?: string | null;

  created_by: string;
  updated_by?: string | null;
  deleted_by?: string | null;

  is_deleted: boolean;

  created_at: string;
  updated_at: string;
  deleted_at?: string | null;
};

export type ListLettersResponse = {
  items: Letter[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
};